src/test/java/org/sagebionetworks/bridge/models/schedules/ActivitySchedulerTest.java

```diff
@@ -454,6 +460,35 @@ public void greaterOfMinOrTimeBasedSchedulingReturned() {
         assertTrue(scheduledActivities.size() > 1);
     }
     
+    @Test
+    public void getScheduleWindowsBasedOnEventsAdjustsTimeZonesForAllSequences() {
+        Map<String,DateTime> events = ImmutableMap.of(
+                "enrollment", NOW, 
+                "enrollment2", NOW.plusWeeks(2)); // both in -07:00 timezone
+        
+        ScheduleContext context = new ScheduleContext.Builder()
+                .withStudyIdentifier(TEST_STUDY)
+                // not the enrollment time zone, and not the request timezone
+                .withInitialTimeZone(DateTimeZone.forOffsetHours(-4)) 
+                .withStartsOn(DateTime.parse("2015-03-26T14:40:00+05:30"))
+                .withEndsOn(DateTime.parse("2015-03-26T14:40:00+05:30"))
+                .withEvents(events).build();
+        
+        assertEquals(context.getRequestTimeZone(), IST);
+        
+        Schedule schedule = new Schedule();
+        schedule.setEventId("enrollment,enrollment2");
+        // Need a recurring schedule with a sequence to generate an end time and multiple windows
+        schedule.setSequencePeriod("P1W");
+        schedule.setScheduleType(RECURRING);
```

Comment:
We should also have a test that uses delay, just to make sure we set the time zone in both cases.